packages/server/modules/core/repositories/branches.ts

```diff
@@ -131,22 +134,59 @@ function getPaginatedProjectModelsBaseQuery<T>(
   return q
 }
 
+export async function getStructuredProjectModels(projectId: string) {
```

Comment:
note edge case discussed with Fabs that makes this worth it: 
we have the following branches: `test/inner`, and `test/super` - branch `test` does not exist => we can't get away easily by just adding a `children` field to the branch model (unless we do a migration? - sounds complicated)